iree/hal/cuda/cuda_device.c

```diff
@@ -229,15 +263,19 @@ static iree_status_t iree_hal_cuda_device_queue_submit(
     iree_hal_queue_affinity_t queue_affinity, iree_host_size_t batch_count,
     const iree_hal_submission_batch_t* batches) {
   iree_hal_cuda_device_t* device = iree_hal_cuda_device_cast(base_device);
-  for (int i = 0; i < batch_count; i++) {
-    for (int j = 0; j < batches[i].command_buffer_count; j++) {
-      CUgraphExec exec = iree_hal_cuda_graph_command_buffer_exec(
-          batches[i].command_buffers[j]);
-      CUDA_RETURN_IF_ERROR(device->context_wrapper.syms,
-                           cuGraphLaunch(exec, device->stream),
-                           "cuGraphLaunch");
+  iree_hal_command_buffer_t* stream_command_buffer;
+  iree_status_t status = iree_hal_cuda_stream_command_buffer_create(
```

Comment:
are we sure we always want to switch like this? I forget, have there been any performance comparisons?

Reply:
I definitely agree with the alternative options, but might need some guidence for the problem I mentioned. actually no performance comparisons

Reply:
if you add a boolean option to iree_hal_cuda_device_params_t for whether to use graph mode then you can store a flag on iree_hal_cuda_device_t and then add an if statement in the command buffer creation and submission to switch between the two (`if (device->use_graph_submission) { return new graph_command_buffer... } else { return new deferred_command_buffer... }` etc)

Reply:
(then we can have a flag in the CUDA device registration like --cuda_use_graph_submission to allow switching on the command line for ease of comparison)

Reply:
I thought you were saying the ways of implementaion lol. I added the check of use_graph_submission, but not sure how to set it by command flag. I left a TODO in cuda/registration/driver_module.c.